src/ddprof_worker.cc

```diff
@@ -790,8 +847,12 @@ DDRes ddprof_worker_process_event(const perf_event_header *hdr, int watcher_pos,
           // Handle special profiling types first
           if (watcher->ddprof_event_type == DDPROF_PWE_tALLOCSYS1 ||
               watcher->ddprof_event_type == DDPROF_PWE_tALLOCSYS2) {
-            DDRES_CHECK_FWD(
-                ddprof_pr_sysallocation_tracking(ctx, sample, watcher_pos));
+            bool clear_pid = false;
+            DDRES_CHECK_FWD(ddprof_pr_sysallocation_tracking(
```

Comment:
This is good for now, but making some irrelevant observations for future development.

Consider when a PID is invalidated in one of these sample-handlers.
1. When a PID is invalidated for the purposes of heap/allocation tracking, it may not be invalid for the purpose of other tracking.
2. When a PID has seen an invalidating event in one ringbuffer, there may be valid events referencing that PID in other ringbuffers.

I think it would be premature to act on these observations now.

Reply:
True. This is not good enough.
Let's talk more about it.